### PR TITLE
Add statistics implementation

### PR DESCRIPTION
**Pull request by okx-code**
# Description
Add implementation for `Player#setStatistic`, `Player#getStatistic` etc

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Version number updated in `settings.gradle`.
- [x] Unit tests added.
- [x] Code follows existing code format.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/statistic/StatisticMock.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.bukkit.Material;
+import org.bukkit.Statistic;
+import org.bukkit.entity.EntityType;
+
+public class StatisticMock
+{
+
+	private final Map<Statistic, Integer> untypedStatistics = new Hashtable<>();
+	private final Map<Statistic, Map<Material, Integer>> materialStatistics = new HashMap<>();
+	private final Map<Statistic, Map<EntityType, Integer>> entityStatistics = new HashMap<>();
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/statistic/StatisticMock.java]: `Statistic` is an enum, so these should all be `EnumMap`s.

File: src/main/java/be/seeseemelk/mockbukkit/statistic/StatisticMock.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	}
+
+	public int getStatistic(Statistic statistic, EntityType entity)
+	{
+		Map<EntityType, Integer> map = entityStatistics.get(statistic);
+		if (map == null) {
+			return 0;
+		}
+		return map.getOrDefault(entity, 0);
+	}
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/statistic/StatisticMock.java]: Some `@NotNull` annotations for parameters would be nice.

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 3]
```diff
@@ -1159,127 +1162,109 @@ public void updateInventory()
...
-		throw new UnimplementedOperationException();
+		statistics.setStatistic(statistic, statistics.getStatistic(statistic) + amount);
 	}
 
 	@Override
 	public void decrementStatistic(@NotNull Statistic statistic, int amount)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		statistics.setStatistic(statistic, statistics.getStatistic(statistic) - amount);
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]: two operations for what can be done in one, I suggest you to add a `increment` and `decrement` method in your `StatisticsMock` and implement the logic there.

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 4]
```diff
@@ -1159,127 +1162,109 @@ public void updateInventory()
...
-		throw new UnimplementedOperationException();
+		statistics.setStatistic(statistic, material, statistics.getStatistic(statistic, material) + amount);
 	}
 
 	@Override
 	public void decrementStatistic(@NotNull Statistic statistic, @NotNull Material material, int amount)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		statistics.setStatistic(statistic, material, statistics.getStatistic(statistic, material) - amount);
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]: two operations for what can be done in one, I suggest you to add a `increment` and `decrement` method in your `StatisticsMock` and implement the logic there.

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 5]
```diff
@@ -1159,127 +1162,109 @@ public void updateInventory()
...
-		throw new UnimplementedOperationException();
+		statistics.setStatistic(statistic, entityType, statistics.getStatistic(statistic, entityType) + amount);
 	}
 
 	@Override
 	public void decrementStatistic(@NotNull Statistic statistic, @NotNull EntityType entityType, int amount)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		statistics.setStatistic(statistic, entityType, statistics.getStatistic(statistic, entityType) - amount);
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]: two operations for what can be done in one, I suggest you to add a `increment` and `decrement` method in your `StatisticsMock` and implement the logic there.

File: src/main/java/be/seeseemelk/mockbukkit/statistic/StatisticMock.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.UUID;
+import org.bukkit.Material;
+import org.bukkit.Statistic;
+import org.bukkit.entity.EntityType;
+
+public class StatisticMock
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/statistic/StatisticMock.java]: Some javadocs would be nice :eyes:

For the class and methods

Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/statistic/StatisticMock.java]: Also shouldn't this be `StatisticsMock` (plural, since it holds multiple).

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Comment by okx-code** [comment]
I've added your suggested improvements.

**Review by TheBusyBiscuit — APPROVED** [review]
There is still a bit room for optimization left but it looks good to me

**Pull request merged**